### PR TITLE
Support loss of .ji in lib

### PR DESCRIPTION
**Pull request by waTeim**
A fix in julia-config that doesn't assume that sys.ji must exist.  Backward compatible

### CODE REVIEW

File: contrib/julia-config.jl [hunk 1]
```diff
@@ -21,9 +21,34 @@ function includeDir()
     joinpath(match(r"(.*)(bin)",JULIA_HOME).captures[1],"include","julia");
 end
 
+function osxInitDir()
+    if imagePath()[end-1:end] == "ji"
+        return return match(r"(.*)(/julia/sys.ji)",imagePath()).captures[1];
```
Comment (tkelman) [inline contrib/julia-config.jl]: return return?

File: contrib/julia-config.jl [hunk 2]
```diff
@@ -21,9 +21,34 @@ function includeDir()
...
+    else
+       return match(r"(.*)(/julia/sys.dylib)",imagePath()).captures[1];
+    end
+end
+
+function linuxInitDir()
+    if imagePath()[end-1:end] == "ji"
+        return return match(r"(.*)(/julia/sys.ji)",imagePath()).captures[1];
+    else
+       return match(r"(.*)(/julia/sys.so)",imagePath()).captures[1];
```
Comment (tkelman) [inline contrib/julia-config.jl]: one too few leading spaces

Comment (ScottPJones) [inline contrib/julia-config.jl]: How does it return return?

**Comment by waTeim** [comment]
Ugh.  all syntax error a typos... updated.

**Comment by tkelman** [comment]
I think this could potentially break on freebsd, since neither `@linux_only` or `@osx_only` apply there.

**Comment by waTeim** [comment]
Yea;  too much cut-and-paste.  See improved vers.

**Comment by tkelman** [comment]
I think this should work. Let us know if any other tweaks are needed.

**Pull request merged**